src/providers/gdal/qgsgdalprovider.cpp

```diff
@@ -2735,11 +2741,10 @@ QGISEXTERN QgsGdalProvider *create(
     return new QgsGdalProvider( uri, error );
   }
 
-  GDALSetGeoTransform( dataset.get(), geoTransform );
-  GDALSetProjection( dataset.get(), crs.toWkt().toLocal8Bit().data() );
-  dataset.reset();
+  GDALSetGeoTransform( dataset, geoTransform );
+  GDALSetProjection( dataset, crs.toWkt().toLocal8Bit().data() );
 
-  return new QgsGdalProvider( uri, true );
+  return new QgsGdalProvider( uri, true, dataset );
```

Comment:
Why not leave the unique_ptr, but just release it here?

Reply:
Re-added per faa0861d97b5c7834180b884a159d224cdac44fa